impeller/renderer/compute_pass.h

```diff
@@ -35,6 +35,25 @@ class ComputePass : public ResourceBinder {
 
   virtual fml::Status Compute(const ISize& grid_size) = 0;
 
+  /// @brief Ensures all previously encoded commands have finished writing to
+  ///        buffers before any following commands can read from those buffer.
+  ///
+  ///        This method after encoding a command that writes to a storage
+  ///        buffer that a subsequent command will read from, provided that
+  ///        is a compute command and not a render pass command. It does not
+  ///        matter if the compute command is in a different command buffer.
+  virtual void AddBufferMemoryBarrier() = 0;
+
+  /// @brief Ensures all previously encoded commands have finished writing to
```

Comment:
_Minor edit for clarity:_ "Ensures that side effects (to textures) of all commands previously encoded in the pass are visible to subsequent commands in the pass."

Since "finished writing" isn't quite the same as having those writes be visible to subsequent commands. Because of caches and such.

Reply:
Done